packages/indexer-serverless/src/functions/producer/handler.ts

```
@@ -35,108 +47,58 @@ export default async function producer() {
   const batchSize = parseInt(process.env.BATCH_SIZE!);
 
   console.log({
+    batchSize,
     lastQueuedEndBlock,
-    targetBlockHeight,
+    currentChainHeight,
     existingLastQueuedEndBlock,
   });
 
-  if (targetBlockHeight < lastQueuedEndBlock) {
-    console.log(`Last queued message is up to the target block height`);
+  if (currentChainHeight < lastQueuedEndBlock) {
+    console.log(`Last queued message is up to the chain height`);
     return;
   }
 
-  const targetJobCount = Math.min(
-    maxBlocksToQueuePerExecution,
-    targetBlockHeight - lastQueuedEndBlock,
-  );
-  const targetLastQueuedEndBlock = lastQueuedEndBlock + targetJobCount;
-
-  const dispatch = async (jobs: BatchSQSMessage[]) => {
-    if (jobs.length) {
-      await sqs
-        .sendMessageBatch({
-          QueueUrl: process.env.QUEUE_URL!,
-          Entries: jobs,
-        })
-        .promise();
-    }
-  };
-
-  while (true) {
-    let batches = batchBlocks(
-      lastQueuedEndBlock === 0 ? 0 : lastQueuedEndBlock + 1,
-      targetLastQueuedEndBlock,
-      batchSize,
-      10,
-    );
-
-    const dispatches = batches.batches.map((b) => {
-      let workerGroup = 0;
-      const blocksInBatch = b.endBlock + 1 - b.startBlock;
-      // For sanity only set a different worker group if the block range in
-      // this batch is the same length as the configured batch size,
-      // otherwise use group 0
-      if (blocksInBatch === batchSize) {
-        const batchGroup = b.startBlock % (maxWorkers * batchSize);
-        workerGroup = batchGroup / maxWorkers;
-      }
-      return {
-        Id: `${b.endBlock}`,
-        MessageBody: JSON.stringify(b),
-        MessageGroupId: `${workerGroup}`,
-      };
-    });
-
-    await dispatch(dispatches);
-
-    lastQueuedEndBlock = batches.lastBlock;
-
-    await saveLastQueuedEndBlock(lastQueuedEndBlock);
-
-    if (lastQueuedEndBlock >= targetLastQueuedEndBlock) {
+  const targetJobCount = maxBlocksToQueuePerExecution / batchSize;
```

might want to throw an error here if the result is a float